### PR TITLE
New server `statistics` endpoint

### PR DESCRIPTION
**Pull request by forman**
Added new statistics API to xcube server. The service computes basic statistical values and a histogram for given data variable, time stamp, and a GeoJSON geometry. Its endpoint is: `/statistics/{datasetId}/{varName}?time={time}`. Geometry is passed as request body in form of a GeoJSON geometry object.

To verify:

1. In xcube project folder, run xcube server with the demo configuration:  `xcube --loglevel=INFO --traceback serve --traceperf -vvv -c examples/serve/demo/config.yml`
2. Open `http://127.0.0.1:8080/openapi.html` to verify the API is there
3. Open `http://127.0.0.1:8080/viewer` and try the new `Statistics`tab in the right sidebar.

The current state of https://github.com/xcube-dev/xcube-viewer/pull/357 has been bundled in this PR.

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Comment by konstntokas** [comment]
Screenshot from 2024-06-04 11-32-05
FYI (not super important): When selecting the range of the histogram, when the upper bound is close to the maximal value, the value disappears at the right edge.

**Comment by forman** [comment]
> When selecting the range of the histogram, when the upper bound is close to the maximal value, the value disappears at the right edge.

Yes, I know. I will fix that, if at all, another time. It requires extra padding to the right if will take away space for the chart.

File: xcube/webapi/statistics/controllers.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+DEFAULT_BIN_COUNT = 100
+
+
+def compute_statistics(
+    ctx: StatisticsContext,
+    ds_id: str,
+    var_name: str,
+    geo_json: dict[str, Any],
+    params: Mapping[str, str],
+):
```
[the remark this answers is not in the document]
Comment (forman) [reply xcube/webapi/statistics/controllers.py:25]: Because it is no API

**Comment by forman** [comment]
> Further, the new functionalities could be added to the docs/source/viewer.rst so the new functionalities regarding statistics are added to the documentation. Maybe we could do that in another PR.

@konstntokas Yes. But we wont do that in the RST. Instead, we'll set up an entirely new and dedicated doc site for the viewer.

**Comment by forman** [comment]
> I will fix that, if at all, another time.

Done!

File: test/webapi/statistics/test_controllers.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        self.assertAlmostEqual(44.5496, result.get("mean"), places=4)
+        self.assertNotIn("histogram", result)
+
+    def test_compute_statistics_for_oor_point(self):
+        lon = -100  # --> out-of-range!
+        lat = 51.465
+        time = "2017-01-16 10:09:21"
+
+        ctx = get_statistics_ctx()
+        dataset = ctx.datasets_ctx.get_dataset("demo")
```
[the remark this answers is not in the document]
Comment (forman) [reply test/webapi/statistics/test_controllers.py]: Done!

**Pull request merged**